pilot/cmd/pilot-agent/main.go

```diff
@@ -305,6 +295,39 @@ var (
 	}
 )
 
+func getPilotSAN(domain string, ns string) []string {
+	var pilotSAN []string
+	if controlPlaneAuthPolicy == meshconfig.AuthenticationPolicy_MUTUAL_TLS.String() {
```

Comment:
Does this function really need to care about the MUTUAL_TLS setup? If MTLS is not enabled, will the Pilot SAN be simply ignored?

Reply:
We extracted the two functions to make them testable and this is the behavior as it was before. Probably, the function doesn't need to take care about controlPlaneAuthPolicy, though the Pilot SAN is not used with controlPlaneAuthPolicy other than MUTUAL_TLS.